frontend/src/app/modules/leader/leader-board/leader-board.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { UserService } from '@core/services/user.service';
+import { IPageSettings } from '@shared/models/page-settings';
+import { IUser } from '@shared/models/user/user';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+    selector: 'app-leader-board',
+    templateUrl: './leader-board.component.html',
+    styleUrls: ['./leader-board.component.sass'],
+})
+export class LeaderBoardComponent implements OnInit {
```

Comment:
I guess you can implement base component as well (it will simplify working with unsubscribe$)
you can find it in app/core/base/